src/app/components/fileupload/fileupload.ts

```diff
@@ -557,11 +557,10 @@ export class FileUpload implements AfterViewInit, AfterContentInit, OnInit, OnDe
 
         this.onSelect.emit({ originalEvent: event, files: files, currentFiles: this.files });
 
-        if (this.fileLimit) {
-            this.checkFileLimit();
-        }
+         // this will check the fileLimit with the uploaded files
+        this.checkFileLimit(files);
 
-        if (this.hasFiles() && this.auto && (!(this.mode === 'advanced') || !this.isFileLimitExceeded())) {
+        if (this.hasFiles() && this.auto && !(this.mode === 'advanced') && !this.isFileLimitExceeded()) {
```

Comment:
This brokes the [auto]="true" functionality for 'advanced' mode (the default one). So by default [auto]="true" will not work. Fix required.